File: PN39.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;  
        	

    public:
        int iCnt = 0;
		
  
		
		void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        int MultFactors()
		{
			
			int iCnt = 0, iSum = 1;
			
			for(iCnt = 1; iCnt <= (iNo/2); iCnt++)
			{
				if((iNo % iCnt) == 0)
				{
					iSum = iSum * iCnt;
				}
					
			}
			return iSum;
     	}
	
};

int main()
{
    Number nobj;
	int iRet = 0;
	


    nobj.Accept();
   

	iRet = nobj.MultFactors();
	cout<<iRet<<endl;
	
	


    return 0;
}

File: PN39.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }

    public int SumDigits()
    {
		

        int iDigit = 0;
        int iSum = 0;
        if(iNo < 0)
        {
           iNo = -iNo;
        }
        while(iNo > 0)
        {
           iDigit = iNo % 10;
		   if((iNo %2) == 0)
		   {
              iSum = iSum + iDigit;
		   }  
           iNo = iNo /10;
        }
        return iSum;
	}	
}

class PN39
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            int iRet = 0;

            nobj.Accept();
            nobj.Display();

            iRet = nobj.SumDigits();
			System.out.println("Value is: "+iRet);
			
            
    }
}